src/arti/tor-dirmgr/src/config.rs

```diff
@@ -190,6 +190,14 @@ impl NetDirConfigBuilder {
         self.cache_path = Some(path.to_path_buf());
     }
 
+    /// Set directory authorities.
+    pub fn set_authorities(&mut self, authorities: &[Authority]) {
+        self.network.authority.clear();
+        for authority in authorities {
+            self.network.authority.push(authority.clone());
+        }
```

Comment:
```suggestion
        self.network.authority = authorities.to_vec();
```

Reply:
I was thinking that could cause problems if the original vector is referred elsewhere, but is that case also prevented by the ownership mechanism?

Reply:
> I was thinking that could cause problems if the original vector is referred elsewhere

[`to_vec` requires that the values implement `Clone`](https://doc.rust-lang.org/std/primitive.slice.html#method.to_vec), so it's pretty much doing what you've been writing (but more efficiently).

> is that case also prevented by the ownership mechanism?

Of course, the only way to break it is to use some `unsafe` code.